cmd/kubectl-k8ssandra/register/register_test.go

```diff
@@ -0,0 +1,178 @@
+package register
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	configapi "github.com/k8ssandra/k8ssandra-operator/apis/config/v1beta1"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestRegister(t *testing.T) {
+	require.New(t)
+	client1 := (*multiEnv)[0].GetClient("source-namespace")
+	client2 := (*multiEnv)[1].GetClient("dest-namespace")
+
+	if err := client1.Create((*multiEnv)[0].Context, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "source-namespace"}}); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := client2.Create((*multiEnv)[1].Context, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "dest-namespace"}}); err != nil {
+		t.Fatal(err)
+	}
+
+	buildDir := os.Getenv("BUILD_DIR")
+	if buildDir == "" {
+		_, b, _, _ := runtime.Caller(0)
+		buildDir = filepath.Join(filepath.Dir(b), "../../../build")
+	}
+
+	if _, err := os.Stat(buildDir); os.IsNotExist(err) {
+		if err := os.Mkdir(buildDir, os.ModePerm); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	kc1, err := (*multiEnv)[0].GetKubeconfig(t)
+	if err != nil {
+		t.Fatal(err)
+
+	}
+	f, err := os.Create(buildDir + "/kubeconfig1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
```

Comment:
question: Why defer in a test (instead of just Close() and testing the output value) ?

Reply:
There's t.Cleanup for some functionality with guaranteed test semantics (defer behaves unlike you expect if used in the tests with t.Parallel)

Reply:
I'll implement this, I wasn't aware of the t.Parallel issue. To be fair, we don't support parallel tests anyway given the static filenames, but maybe close()  is a more idiomatic solution indeed..